test/test_nestedtensor.py

```diff
@@ -0,0 +1,146 @@
+import torch
+import unittest
+from common_utils import TEST_WITH_PROTOTYPES
+
+if TEST_WITH_PROTOTYPES:
+    import torch.prototypes.nestedtensor as nestedtensor
+    from common_utils import TestCase
+    NestedTensor = nestedtensor.NestedTensor
+else:
+    print('prototypes not available, skipping tests')
+    TestCase = object
+
+
+def _shape_prod(shape_):
+    shape = tuple(shape_)
+    start = 1
+    for s in shape:
+        start = start * s
+    return start
+
+# From torchaudio by jamarshon
+def random_float_tensor(seed, size, a=22695477, c=1, m=2 ** 32):
+    """ Generates random tensors given a seed and size
+    https://en.wikipedia.org/wiki/Linear_congruential_generator
+    X_{n + 1} = (a * X_n + c) % m
+    Using Borland C/C++ values
+     The tensor will have values between [0,1)
+    Inputs:
+        seed (int): an int
+        size (Tuple[int]): the size of the output tensor
+        a (int): the multiplier constant to the generator
+        c (int): the additive constant to the generator
+        m (int): the modulus constant to the generator
+    """
+    num_elements = 1
+    for s in size:
+        num_elements *= s
+
+    arr = [(a * seed + c) % m]
+    for i in range(num_elements - 1):
+        arr.append((a * arr[i] + c) % m)
+
+    return torch.tensor(arr).float().view(size) / m
+
+
+def random_int_tensor(seed, size, low=0, high=2 ** 32, a=22695477, c=1, m=2 ** 32):
+    """ Same as random_float_tensor but integers between [low, high)
+    """
+    return torch.floor(random_float_tensor(seed, size, a, c, m) * (high - low)) + low
+
+
+class TestNestedTensor(TestCase):
+
+    def gen_float_tensor(self, seed, shape):
+        return random_float_tensor(seed, shape)
```

Comment:
What's wrong with just doing a torch.manual_seed and then using torch.rand?

Reply:
It's not deterministic. The order determines the outcome. If I call into this with the same seed, I'll always get the same output. That also means it's very easy to debug and reproduce errors. With a manual seed and subsequent calls to random I'm not in control of exactly what sequence I'm using.

Reply:
What if you always manual seed in this function? So 
```
def random_float_tensor(seed, size):
    torch.manual_seed(seed)
    return torch.rand(size)
```
this should always be deterministic because you're always seeding (with the same seed) before calling rand.

Reply:
You're modifying a global variable for local effect and you're not invariant under the used RNG